features/events/daemon/src/main/java/org/opennms/netmgt/eventd/BroadcastEventProcessor.java

```diff
@@ -147,7 +148,8 @@ private boolean isReloadConfigEvent(Event event) {
         boolean isTarget = false;
         
         if (EventConstants.RELOAD_DAEMON_CONFIG_UEI.equals(event.getUei())) {
-            if ("Eventd".equalsIgnoreCase(AbstractEventUtil.getInstance().getValueOfParm(EventConstants.PARM_DAEMON_NAME, event))) {
+            Parm target = event.getParm(EventConstants.PARM_DAEMON_NAME);
+            if (target != null && "Eventd".equalsIgnoreCase(target.getValue().getContent())) {
```

Comment:
> On Apr 12, 2016, at 8:47 AM, Alejandro Galue <notifications@github.com> wrote:
> 
> >          boolean isTarget = false;
> >
> >          if (EventConstants.RELOAD_DAEMON_CONFIG_UEI.equals(event.getUei())) {
> > -            if ("Eventd".equalsIgnoreCase(AbstractEventUtil.getInstance().getValueOfParm(EventConstants.PARM_DAEMON_NAME, event))) {
> > +            Parm target = event.getParm(EventConstants.PARM_DAEMON_NAME);
> > +            if (target != null && "Eventd".equalsIgnoreCase(target.getValue().getContent())) {
> 
> That is theoretically impossible. In fact, if that can be null, there are hundreds of lines of codes in OpenNMS that would require a NPE check.

It is *in fact* impossible since a boolean is a primitive.

:D